src/flick/lst/views.py

```diff
@@ -1,3 +1,160 @@
+from rest_framework import generics, mixins, status, viewsets
+from rest_framework.response import Response
+
+from .models import Lst
+from .serializers import LstSerializer
+from api import settings as api_settings
+from api.utils import failure_response, success_response
+from show.models import Show
+from user.models import Profile
+
+import json
+import re
+
+
+class LstList(generics.ListCreateAPIView):
+
+    queryset = Lst.objects.all()
+    serializer_class = LstSerializer
+
+    permission_classes = api_settings.CONSUMER_PERMISSIONS
+
+    def list(self, request):
+        """See all possible lists."""
+        # TODO: only allow users who are owners / collaborators see lists
+        self.serializer_class = LstSerializer
+        return super(LstList, self).list(request)
+
+    def post(self, request):
+        """Create a list."""
+        data = json.loads(request.body)
+        lst_name = data.get("lst_name")
+        lst_pic = data.get("lst_pic")
+        is_favorite = data.get("is_favorite", False)
+        is_private = data.get("is_private", False)
+        is_watched = data.get("is_watched", False)
+        collaborator_ids = data.get("collaborators", [])
+        show_ids = data.get("shows", [])
+
+        lst = Lst()
+        lst.lst_name = lst_name
+        lst.lst_pic = lst_pic
+        lst.is_favorite = is_favorite
+        lst.is_private = is_private
+        lst.is_watched = is_watched
+        profile = Profile.objects.get(user=request.user)
+        lst.owner = profile
+        lst.save()
+
+        for c_id in collaborator_ids:
+            collaborator = User.objects.get(pk=c_id)
+            if Profile.objects.filter(user=collaborator):
+                c = Profile.objects.get(user=collaborator)
+                lst.collaborators.add(c)
+        for show_id in show_ids:
+            if Show.objects.filter(pk=show_id):
+                show = Show.objects.get(pk=show_id)
+                lst.shows.add(show)
+        lst.save()
+        serializer = LstSerializer(lst)
+        return success_response(serializer.data)
+
+
+class LstDetail(generics.GenericAPIView):
+
+    queryset = Lst.objects.all()
+    serializer_class = LstSerializer
+
+    permission_classes = api_settings.UNPROTECTED
+
+    def get(self, request, pk):
+        """Get a specific list by id."""
+        # TODO: only allow users who are owners / collaborators see lists
+        queryset = self.get_object()
+        serializer = LstSerializer(queryset, many=False)
+        return success_response(serializer.data)
+
+    def delete(self, request, pk):
+        """Delete a list by id."""
+        # TODO: only allow users who are owners delete a list
+        if not Lst.objects.filter(pk=pk):
+            return success_response(f"List of id {pk} has already been deleted.")
+        lst = Lst.objects.get(pk=pk)
+        lst.delete()
+        return success_response(f"List of id {pk} has been deleted.")
+
+    def post(self, request, pk):
+        """
```

Comment:
what does is watched mean for a list?

Reply:
whether or not this list is a list full of shows that you've already watched!